spec/rspec/expectations/handler_spec.rb

```diff
@@ -209,6 +224,25 @@ module Expectations
 
           RSpec::Expectations::NegativeExpectationHandler.handle_matcher(actual, matcher, failure_message)
         end
+
+        it "add custom failure message to mock matcher if there is one" do
+          error_generator = instance_double("RSpec::Mocks::ErrorGenerator", :opts => { :message => {} })
+          verifying_message_expectation = instance_double(
+            "RSpec::Mocks::VerifyingMessageExpectation",
+            :error_generator => error_generator
+          )
+          matcher = double(
+            "matcher",
+            :failure_message_when_negated => "custom",
+            :does_not_match? => verifying_message_expectation
+          )
+
+          actual = Object.new
+
+          handle_matcher_result = RSpec::Expectations::NegativeExpectationHandler.handle_matcher(actual, matcher, "custom")
+
+          expect(handle_matcher_result.error_generator.opts).to eq({ :message => "custom" })
```

Comment:
Does it have to be explicitly wrapped in curly braces?

Reply:
I did this for old Rubies if I remember.